cpp/support/thread_safe_cache.h

```diff
@@ -179,6 +188,228 @@ class ThreadSafeCache<Key, Value> {
   std::shared_mutex erase_mutex_;
 };
 
+namespace details {
+
+template <typename Value, typename... Keys>
```

Comment:
```suggestion
template <typename Keys, typename Value>
```
don't consider multiple keys in thread_safe_cache.h